### PR TITLE
network: nexthop: tiny cleanups

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18589.

### CODE REVIEW

File: src/network/networkd-nexthop.c [hunk 1]
```diff
@@ -300,7 +300,7 @@ static int nexthop_configure(NextHop *nexthop, Link *link) {
 
         link_ref(link);
 
-        r = nexthop_add(link, nexthop, &nexthop);
+        r = nexthop_add(link, nexthop, NULL);
```
Comment (bluca) [inline src/network/networkd-nexthop.c:303]: commit message says "network: constify arguments" but this is removing a parameter? what's the reason for this change?

Comment (yuwata) [reply src/network/networkd-nexthop.c:303]: As it is not necessary to update the pointer `nexthop`. The function returns soon later without using `nexthop` variable.

Comment (bluca) [reply src/network/networkd-nexthop.c:303]: I see, thanks

**Comment by keszybz** [comment]
centosci:
```
16:13:38 ======================================================================
16:13:38 FAIL: test_nexthop (__main__.NetworkdNetworkTests)
16:13:38 ----------------------------------------------------------------------
16:13:38 Traceback (most recent call last):
16:13:38   File "/build/test/test-network/systemd-networkd-tests.py", line 2803, in test_nexthop
16:13:38     self.assertIn('id 1 via 192.168.5.1 dev veth99', output)
16:13:38 AssertionError: 'id 1 via 192.168.5.1 dev veth99' not found in 'id 1 via 192.168.5.2 dev veth99 scope link proto static \nid 2 via 2001:1234:5:8f63::2 dev veth99 scope link proto static \nid 3 dev veth99 scope link proto static \nid 4 dev veth99 scope host proto static'
16:13:38 
16:13:38 ----------------------------------------------------------------------
```

Looks relevant.

File: src/network/networkd-nexthop.c [hunk 2]
```diff
@@ -406,9 +406,15 @@ int manager_rtnl_process_nexthop(sd_netlink *rtnl, sd_netlink_message *message,
...
         r = sd_netlink_message_read_u32(message, NHA_ID, &tmp->id);
-        if (r < 0 && r != -ENODATA) {
+        if (r == -ENODATA) {
+                log_link_warning_errno(link, r, "rtnl: received nexthop message without NHA_ID attribute, ignoring: %m");
+                return 0;
+        } else if (r < 0) {
                 log_link_warning_errno(link, r, "rtnl: could not get NHA_ID attribute, ignoring: %m");
                 return 0;
+        } else if (tmp->id == 0) {
+                log_link_warning_errno(link, r, "rtnl: received nexthop message with invalid nexthop ID, ignoring: %m");
```
Comment (poettering) [inline src/network/networkd-nexthop.c]: second argument should probably be explicitly set to zero, given that r doesn#t really carry an error here, but doesn't matter, since it's gonna be 0 here in that case anyway. but still looks like a bug, even though it isn't

**Comment by poettering** [comment]
lgtm, besides the nitpick and the CI issue

**Comment by yuwata** [comment]
@poettering and @keszybz Thank you for the comments. The logging issue is addressed. The CI failure should be fixed already merged commit 055e4909de8f83563e3e8bba1ac64db80c916bb4. PTAL.
